segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3Storage.java

```diff
@@ -293,7 +293,13 @@ private SegmentProperties syncCreate(String streamSegmentName) throws StreamSegm
          * But this does not work. Currently all the calls to putObject API fail if made with reqest.setIfNoneMatch("*").
          * once the issue with extended S3 API is fixed, addition of this one line will ensure put-if-absent semantics.
          * See: https://github.com/pravega/pravega/issues/1564
+         *
+         * This issue is fixed in some versions of extended S3 implementation. The following code sets the IfNoneMatch
+         * flag based on configuration.
          */
+        if (config.isUseNoneMatch()) {
+            request.setIfNoneMatch("*");
```

Comment:
The exception flows through the same way other exceptions do. (through `handleException` and `translateException`). The exception in this case is `PreConditionFailedException` which is translated to `StreamSegmentExistsException`.